src/crc32.rs

```diff
@@ -42,6 +42,13 @@ pub trait Hasher32 {
 pub fn update(mut value: u32, table: &[u32; 256], bytes: &[u8], calc: &CalcType) -> u32 {
     match calc {
         CalcType::Normal => {
+            value = !value;
```

Comment:
@inferiorhumanorgans  Is this intended to change how `Normal` works?

@CLomanno looks like reflect value in and out doesn't affect calculation result?

Reply:
Yes the intent is to allow both reflected and non-reflected CRCs.  Perhaps it would be better to rename the values to None, ReflectIn, ReflectOut, ReflectBoth?

Reply:
I'm intrigued/concerned that changing `Normal` implementation doesn't require updates on parameters or tests.

If we'd like to support more customization, we could limit `CalcType` to `Normal`/`Reverse` and take `ref_in`, `ref_out`, and `xor_out`(currently `final_xor`) as options. @inferiorhumanorgans @CLomanno  Does this make sense?

Resources:
- http://crccalc.com/
- https://www.autosar.org/fileadmin/user_upload/standards/classic/4-3/AUTOSAR_SWS_CRCLibrary.pdf
- https://www.mathworks.com/help/comm/ref/generalcrcgenerator.html

Even better, width can be taken as parameter to avoid code duplication between crc16/32/64 but is not necessary for this PR.

Reply:
@mrhooray @inferiorhumanorgans @CLomanno 
I have created #45 to add the capability to set ref_in / ref_out individually, in case you all want to take a look and leave your thoughts there. Perhaps we could make the change to avoid code duplication under that issue as well?